src/main/resources/com/puppycrawl/tools/checkstyle/grammar/java.g

```diff
@@ -1384,12 +1383,30 @@ multiplicativeExpression
     :    unaryExpression ((STAR^ | DIV^ | MOD^ ) unaryExpression)*
     ;
 
+constantExpression
+    : IDENT
+    | constant
+    ;
+
+switchExpressionCasesGroup
+    :   ("case"^ (constantExpression) (COMMA constantExpression)*) LAMBDA^ expression SEMI
+    |   "default"^ LAMBDA^ statement
+    ;
+
+// Extended switch expression from Java12
+switchExpression
```

Comment:
@arteam I am not finding anything like switch expression in the JLS 12.
https://docs.oracle.com/javase/specs/jls/se12/html/
Do you know where we can find official documentation on it and it's structure?

Reply:
The JDK commit http://hg.openjdk.java.net/jdk/jdk/rev/de411d537aae#l4.43 mentions jls section 15.29, but I don't see it published either.

Reply:
Thanks for the URL.

So this means switch expressions are technically unfinished and could change once the final version is released?

> mentions the JLS section 15.29, but I don't see it published either.

How are you validating then that your changes are correct then?